### PR TITLE
Fix tail -<number> does not work

### PR DESCRIPTION
**Pull request by Smicry**
echo "a\nb" | ./target/debug/tail -1 
```
b
```

### CODE REVIEW

**Comment by sylvestre** [comment]
Looks great. Could you please also add high level tests in:
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_tail.rs

And make sure that the coverage is great!

**Comment by sylvestre** [comment]
clippy is also complaining with:
```

error: field assignment outside of initializer for an instance created with Default::default()
   --> src/uu/tail/src/tail.rs:104:9
    |
104 |         settings.follow = matches.is_present(options::FOLLOW);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D clippy::field-reassign-with-default` implied by `-D warnings`
note: consider initializing the variable with `Settings { follow: matches.is_present(options::FOLLOW), ..Default::default() }` and removing relevant reassignments
   --> src/uu/tail/src/tail.rs:102:9
    |
102 |         let mut settings: Settings = Default::default();
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#field_reassign_with_default

error: this expression borrows a reference (`&Settings`) that is immediately dereferenced by the compiler
   --> src/uu/tail/src/tail.rs:200:41
    |
200 |             unbounded_tail(&mut reader, &settings);
    |                                         ^^^^^^^^^ help: change this to: `settings`
    |
    = note: `-D clippy::needless-borrow` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

error: this expression borrows a reference (`&Settings`) that is immediately dereferenced by the compiler
   --> src/uu/tail/src/tail.rs:232:41
    |
232 |                 bounded_tail(&mut file, &settings);
    |                                         ^^^^^^^^^ help: change this to: `settings`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

error: this expression borrows a reference (`&Settings`) that is immediately dereferenced by the compiler
   --> src/uu/tail/src/tail.rs:239:45
    |
239 |                 unbounded_tail(&mut reader, &settings);
    |                                             ^^^^^^^^^ help: change this to: `settings`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

error: this expression borrows a reference (`&Settings`) that is immediately dereferenced by the compiler
   --> src/uu/tail/src/tail.rs:248:34
    |
248 |         follow(&mut readers[..], &settings);
    |                                  ^^^^^^^^^ help: change this to: `settings`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

error: could not compile `uu_tail` due to 5 previous errors
```

**Comment by Smicry** [comment]
> Looks great. Could you please also add high level tests in: https://github.com/uutils/coreutils/blob/master/tests/by-util/test_tail.rs
> 
> And make sure that the coverage is great!

ok ~

**Comment by Smicry** [comment]
Fixed #2507

File: tests/by-util/test_tail.rs [hunk 1]
```diff
@@ -358,6 +358,16 @@ fn test_positive_lines() {
         .stdout_is("c\nd\ne\n");
 }
 
+/// Test for reading all but the first NUM lines: `tail -3`.
+#[test]
+fn test_obsolete_syntax() {
+    new_ucmd!()
+        .args(&["-3"])
```
Comment (sylvestre) [inline tests/by-util/test_tail.rs:365]: please add a test when the file is too small (compared to the -X)

Comment (Smicry) [reply tests/by-util/test_tail.rs:365]: Got it.

**Pull request merged**